deploy/lambda/auto-deployment.js

```diff
@@ -0,0 +1,116 @@
+const https = require('https');
+
+exports.handler = async function (event, context) {
+    const autoDeployUsername = process.env.AUTO_DEPLOY_USERNAME;
+    const autoDeployToken = process.env.AUTO_DEPLOY_TOKEN;
+
+    if (!process.env.AUTO_DEPLOY_ENABLED) {
+        console.log('Auto Deploying disabled');
+        return;
+    }
+    if (!autoDeployUsername) {
+        console.error('No auto deploy username was provided!');
+        return;
+    }
+    if (!autoDeployToken) {
+        console.error('No auto deploy token was provided!');
+        return;
+    }
+
+    try {
+        const versionURL = `${process.env.PROD_URL}/version`;
+        const mainBranchURL = 'https://api.github.com/repos/atlassian/github-for-jira/branches/main';
+
+        const deployedCommitSHA = (await getRequest(versionURL)).commit;
```

Comment:
This is checking the deployed production commit sha, so if code is deployed to stage, but we haven't manually deploy to prod yet, this code will keep deploying to stage env. Could be a bit of wastes of compute resources.

Reply:
How is it going to keep deploying to staging? This lambda only runs in prod.

Reply:
No, it won't. This lambda will be zipped and uploaded as an artefact to **PROD** environment only(which is done in [bitbucket repo](https://bitbucket.org/atlassian/github-for-jira-deployment/pull-requests/40)). This lambda will run only for the **PROD**  environment and will only run pipelines for the `main` branch of GH4J. 
Deploying to staging or dev will have no effect on this lambda.

Reply:
All good thanks guys.